src/mod/internal/login_mod.cpp

```diff
@@ -27,6 +27,31 @@
 #include "core/RDP/slowpath.hpp"
 #include "RAIL/client_execute.hpp"
 
+#include <string>
+#include <tuple>
+
+namespace
+{
+    inline std::tuple<std::string, std::string> rpartition(std::string text,
+                                                           const char* seps) {
+        size_t sep_pos = text.find_last_of(seps);
+        if (sep_pos != std::string::npos) {
+            return {text.substr(0, sep_pos), text.substr(sep_pos + 1)};
+        }
+        return {std::string(), text};
+    }
+
+    inline std::string concat_target_login(std::string login, std::string target) {
+        if (target.empty()) {
+            return login;
+        }
+        size_t sep_pos = target.find_last_of("+:");
```

Comment:
Why `rpartition` has a `seps` separator, but not `concat_target_login` ?

Reply:
`rpartition` can be seen as a generic function whereas `concat_target_login` is specific to concat a target with a login.